README.md

```diff
@@ -1,10 +1,64 @@
-# default-SEM-model
+# model_seg_rat_axon-myelin_sem
+---
+
+## Model overview
+![image of segmentation obtained from this model](sem_model_preview.png)
+
 AxonDeepSeg default SEM model and testing image. This model works at a resolution of 0.1 micrometer per pixel and was trained on rat spinal cord data collected via a Scanning Electron Microscope (SEM).
 
+## Dependencies
+- [ivadomed](https://ivadomed.org/) commit: e6554281d07b5afef9e68ce8b02e86b02bd68363
+- [axondeepseg](https://axondeepseg.readthedocs.io/en/latest/) commit: 805868e39eddf620c9f3d60d313cadffb1121bfb
+
+## Segment (ADS)
+To segment an image using this model, use the following command in an `axondeepseg` virtual environment:
+```
+axondeepseg -t SEM -i <IMG_PATH> -s <PIXEL_SIZE>
+```
+The `-m` option can be omitted in this case because this is a default built-in model.
+
+## Train and test (ivadomed)
+This model was trained and tested with `ivadomed` on commit **e65542**. We recommand to install ivadomed in a separate virtual environment to reproduce the original training steps. Inside your venv, make sure you are on the right commit:
```

Comment:
Sorry, I think I spoke too fast earlier.
The commit on which the model was trained is [55fc2067](https://github.com/ivadomed/ivadomed/commit/55fc2067cbb9c97a711e32cf8b5a377fb6d517be) as mentioned in the original [README](https://github.com/axondeepseg/default-SEM-model/blob/main/README.md).

I'm not sure what the dependencies at line 10-11 represent? Are they the commits of ADS and ivadomed when we push the model in production?

Reply:
Ah, yes I forgot to mention this. I actually bumped the ivadomed commit to [e65542](https://github.com/ivadomed/ivadomed/commit/e6554281d07b5afef9e68ce8b02e86b02bd68363) which corresponds to the 2.9.3 release. It is indeed the commit of ivadomed we are using in ADS right now.

I only tested the training with this commit, and I assume @mathieuboudreau did too? I actually don't think training would work with this old commit because, from what I understood, the "micr" subfolders were not yet supported and the SEM dataset is up to date with the newest BIDS version. I am not entirely sure and maybe we should test it with the old commit to see if it works, but so far I think we need to either keep this newer ivadomed commit or change the dataset commit.

Reply:
Ok I understand you point, we just are trying to do different things 😄. We just need to be on the same page on WHAT we want to document here.

So the question is do we want to either:
- document how to train the model TODAY
- document how the model WAS trained

My understanding of this README is to document exactly how the model WAS trained, this is why in the original README template, I put the original commit for both ivadomed (55fc2067) and the dataset (1cddcc6).

Reply:
> Ok so apparently training the model on [55fc20](https://github.com/ivadomed/ivadomed/commit/55fc2067cbb9c97a711e32cf8b5a377fb6d517be) does work even with the "micr" subfolders, which I don't really understand given its not in the bids config file in this commit

Yeah that is weird, even the "force-index" of the files [here](https://github.com/ivadomed/ivadomed/blob/55fc2067cbb9c97a711e32cf8b5a377fb6d517be/ivadomed/loader/bids_dataframe.py#L94) did not account for "micr" at the time.

Reply:
> Ok I understand you point, we just are trying to do different things 😄. We just need to be on the same page on WHAT we want to document here.
> 
> So the question is do we want to either:
> 
> * document how to train the model TODAY
> * document how the model WAS trained
> 
> My understanding of this README is to document exactly how the model WAS trained, this is why in the original README template, I put the original commit used for training for both ivadomed (55fc2067) and the dataset (1cddcc6).

I don't think either exactly is what needed. I would prefer:

* document a demo of how **a** model **was** trained.

I think continuously updating how to train a model with the current versions would be difficult, and documenting how each model was trained may also create a bigger burden on us than needed. As long as there is an example that a user can follow that is not significantly different than how you would train a new model or re-train the model with the current version (where sure they might encounter a bug or two, but could reach out to IVADOMED or ADS for help), would be sufficient.

Reply:
If we wanted or needed to "document" how each model was trained, then at that point I would say we would need to consider containerizing them (e.g. in a Docker or Singularity container), but I don't know if we **really** need this yet at our stage.

Reply:
> I don't think either exactly is what needed. I would prefer:
> * document a demo of how **a** model **was** trained.

I kind of disagree with that. How to train **a** model is already covered in the microscopy tutorial in ivadomed doc. In that case, duplicating the information here seems just more work and prone to error/discrepancies.

> I think continuously updating how to train a model with the current versions would be difficult

I agree, keeping this up-to-date is burdensome and would not serve any purposes in my opinion.

> and documenting how each model was trained may also create a bigger burden on us than needed. As long as there is an example that a user can follow that is not significantly different than how you would train a new model or re-train the model with the current version (where sure they might encounter a bug or two, but could reach out to IVADOMED or ADS for help), would be sufficient.

I think that for reproducibility purposes, the least we can do today (without docker or singularity implementation) is to document the original commits. It is a fairly small burden to document on the person doing the training, but can save a lot of time in the future if we have questions about how the training was done (this happens to us before with the SEM model trained on Keras, and we were never able to reproduces the results).

Reply:
> I kind of disagree with that. How to train **a** model is already covered in the microscopy tutorial in ivadomed doc. In that case, duplicating the information here seems just more work and prone to error/discrepancies.

Most of the information here is basially just that. So should we simply provide the link to the data and the commit hash in the readme, and point all instructions to the IVADOMED? What is the purpose of duplicating the instructions here, if as you say the information here is prone to discrepancies over time.

Reply:
It seems if the purpose of documenting the README is to provide sufficient information to reproduce the training, and we want to avoid duplicating information on how to train a model from what is already in IVADOMED, then we should only provide the very minimal essential information here, correct?

Reply:
> Most of the information here is basically just that. So should we simply provide the link to the data and the commit hash in the readme, and point all instructions to the IVADOMED?

For reproducibility, I like the readme as it is now, it tells the user how to reproduce the training from the config file that is embedded in the model and which field to adapt (path, gpu_id, etc), I would just change to the real commits used for training. Adding the link to the microscopy tutorial would be great as well.

Another approach is to go minimal like you said and just mention the ivadomed and data version the model was trained on.
Here are some other examples in ivadomed repos:
- https://github.com/ivadomed/model_seg_sctumor-edema-cavity_t2-t1_unet3d-multichannel
- https://github.com/ivadomed/model_seg_exvivo_gm-wm_t2_unet2d-multichannel-softseg
- https://github.com/ivadomed/mice_uqueensland_gm

> What is the purpose of duplicating the instructions here, if as you say the information here is prone to discrepancies over time.

Sorry, that was not clear, I just meant that we should not keep the commits/versions documented in this repo up-to-date each time we update ivadomed or ADS.